### PR TITLE
Update gradle config & dependencies

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

**Comment by Goooler** [comment]
```
android.databinding.incremental=true
kapt.incremental.apt=true
```
These two properties have been enabled  by default, could be removed.

image
image

File: build.gradle.kts [hunk 1]
```diff
@@ -10,16 +10,15 @@ plugins {
 buildscript {
     repositories {
         google()
-        jcenter()
```
Comment (topjohnwu) [inline build.gradle.kts:13]: Removing jcenter will not work properly because `koin` v2 is only available on jcenter.

Comment (Goooler) [reply build.gradle.kts:13]: Koin have not been `classpath`ed in `buildscript`'s dependencies, subproejcts `implementation` it with jcenter.

Comment (Goooler) [reply build.gradle.kts:13]: And I just moved `subprojects` repositories into `allprojects` closure

image

**Comment by Goooler** [comment]
@topjohnwu Gradle 7.0 is a big release, I think we can merge this PR now.

**Comment by topjohnwu** [comment]
I believe this is already addressed in some changes

**Pull request closed without merging**